src/BenchmarkDotNet/Diagnosers/ExceptionDiagnoser.cs

```diff
@@ -32,15 +35,23 @@ public void Handle(HostSignal signal, DiagnoserActionParameters parameters) { }
 
         public IEnumerable<Metric> ProcessResults(DiagnoserResults results)
         {
-            yield return new Metric(ExceptionsFrequencyMetricDescriptor.Instance, results.ExceptionFrequency);
+            var descriptor = ExceptionsFrequencyMetricDescriptor.Instance;
+
+            if (descriptor is ExceptionsFrequencyMetricDescriptor concreteDescriptor)
+            {
+                concreteDescriptor.SetConfiguration(Config);
+            }
```

Comment:
It is not a good practice to modify shared state. You should rather inject the flag into a new metric descriptor instance.

Reply:
> It is not a good practice to modify shared state. You should rather inject the flag into a new metric descriptor instance.

Ok,
What about a wrapper class that will take the Instance and the Config.
`public static IMetricDescriptor DescriptorWrapper(IMetricDescriptor instance, Config config)`

Reply:
I don't think that's necessary. Just modify the existing descriptor to not use a singleton.